Improved guess-encoding.py code and workflow (quality score heuristics unchanged)

cviner opened the pull request:
I have made a few minor improvements to [`guess-encoding.py`](https://github.com/brentp/bio-playground/blob/master/reads-utils/guess-encoding.py). I outline the main changes below:

- Permit input from a file provided as an argument, maintaining the ability to input via STDIN.
- Transparently permit input of a gzipped FASTQ, when provided with a FASTQ filename ending in `.gz`.
- Automatically take only quality scores from the input FASTQ, obviating `awk` pre-processing.
- Enforce [PEP8](https://www.python.org/dev/peps/pep-0008/) compliance.
- Add the appropriate shebang.

Please let me know if you would like me to squash some commits, if you require backward compatibility WRT quality score input, or would like me to make any other changes to these commits.

brentp commented:
I wrote this with it in mind that you could do e.g.

```
samtools view $bam | cut -f $quality_col | python guess-encoding.py
```

so I'd prefer to leave that part untouched.

can you revert that part?

I'm happy to have the pep8 and python3 stuff.
Can you squash these into a single commit?
thanks.

cviner commented:
@brentp Thanks for taking a look at this so quickly!

Would you mind if I instead added code to detect if only quality scores are provided and leave in the ability to automatically only inspect every 4th line for FASTQ input? Then both workflows could work, without any `awk` filtering.

I will squash everything as well.

brentp commented:
no. I'd prefer to leave it so that it accepts only quality scores. the burden of using awk is quite minor.

cviner commented:
@brentp Okay, thanks. I have now removed that functionality—quality scores alone are now accepted, as before.

I made some additional minor changes to the top-level documentation and have squashed everything down to a single commit.

Please let me know if you need me to change anything else.

cviner commented:
I have also just refactored my previous changes to control flow with respect to encountering an error. I have fixed them to only exit with a non-zero exit code when encountering an error and have added comments to indicate the reasons for the other `if`/`break` control flow segments.

I have squashed these changes back into the single commit. I have run multiple tests and it appears to function exactly as the previous version. Please let me know if you need me to change anything.

brentp commented:
thanks

cviner commented:
@david-wb Great. I will add a commit and PR to change it to be comma-delimited and maintain three tab-delimited columns, unless @brentp expresses a different preference.

david-wb commented:
Sounds good

brentp commented:
sounds good.
